internal/graph/graph.go

```diff
@@ -633,9 +633,15 @@ func (n *Node) addSample(dw, w int64, labels string, numLabel map[string][]int64
 	if format == nil {
 		format = defaultLabelFormat
 	}
-	for key, nvals := range numLabel {
-		for _, v := range nvals {
-			t := numericTags.findOrAddTag(format(v, key), key, v)
+	for k, nvals := range numLabel {
+		units := numUnit[k]
+		for i, v := range nvals {
+			var t *Tag
+			if len(units) == len(nvals) {
```

Comment:
It's a bit weird that to detect the presence of units we still check the equality of the len of the arrays here while in other places we check for zero like we discussed earlier. Should we be consistent?